cognite/client/_api/transformations/schema.py

```diff
@@ -34,17 +42,80 @@ def retrieve(
+                >>> columns = c.transformations.schema.retrieve(destination=TransformationDestination.instances(data_model, "InstanceSpace")
+
+
+            Get the schema for a transformation producing instance relationship data model::
+
+                >>> from cognite.client import CogniteClient
+                >>> from cognite.client.data_classes import TransformationDestination
+                >>> from cognite.client.data_classes.transformations.common import DataModelInfo
+                >>> c = CogniteClient()
+                >>> data_model = DataModelInfo(space="dataModelSpace", external_id="dataModelExternalId",version="dataModelVersion",destination_type="type", destination_relationship_from_type="relationshipFromType")
+                >>> columns = c.transformations.schema.retrieve(destination=TransformationDestination.instances(data_model, "InstanceSpace")
+
         """
 
-        url_path = utils._auxiliary.interpolate_and_url_encode(self._RESOURCE_PATH + "/{}", str(destination.type))
-        filter = destination.dump(True)
-        filter.pop("type")
-        other_params = {"conflictMode": conflict_mode} if conflict_mode else None
-
-        return self._list(
-            list_cls=TransformationSchemaColumnList,
-            resource_cls=TransformationSchemaColumn,
-            method="GET",
-            resource_path=url_path,
-            filter=other_params,
-        )
+        if destination.type == "nodes" or destination.type == "edges" or destination.type == "instances":
+            url_path = utils._auxiliary.interpolate_and_url_encode(self._RESOURCE_PATH + "/{}", "instances")
+            filter = destination.dump(True)
+            other_params = (
+                {
+                    "conflictMode": conflict_mode,
+                    "withInstanceSpace": with_instance_space,
+                    "isConnectionDefinition": is_connection_definition,
+                    "instanceType": instance_type,
+                }
+                if conflict_mode or with_instance_space or is_connection_definition or instance_type
+                else None
+            )
+
+            return self._list(
+                list_cls=TransformationSchemaColumnList,
+                resource_cls=TransformationSchemaColumn,
+                method="GET",
+                resource_path=url_path,
+                filter=other_params,
+            )
+
+        else:
+            url_path = utils._auxiliary.interpolate_and_url_encode(self._RESOURCE_PATH + "/{}", str(destination.type))
+            filter = destination.dump(True)
+            filter.pop("type")
+            other_params = {"conflictMode": conflict_mode} if conflict_mode else None
+
+            return self._list(
+                list_cls=TransformationSchemaColumnList,
+                resource_cls=TransformationSchemaColumn,
+                method="GET",
+                resource_path=url_path,
+                filter=other_params,
+            )
```

Comment:
This is identical for both branches of the conditional - can be moved out.

Reply:
Resolved.

Line 105

```
 def _retrieve(
        self, destination: TransformationDestination, other_params: Optional[dict]
    ) -> TransformationSchemaColumnList:
```